Vulnerability scan failing for certain images

- Vulnerability scan failing for image `piesecurity/apache-struts2-cve-2017-5638`
```
panic: runtime error: slice bounds out of range [31:29]

goroutine 67 [running]:
github.com/deepfence/vulnerability_mapper/utils.removeFromSlice(...)
        /go/vulnerability_mapper/utils/utils.go:72
github.com/deepfence/vulnerability_mapper/utils.ExtractExploitPOCURL({0xc00031f080?, 0x1f?, 0xc000134ed0?})
        /go/vulnerability_mapper/utils/utils.go:65 +0x22d
github.com/deepfence/vulnerability_mapper/scanner/grype.PopulateFinalReport({_, _, _}, {{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, {0x0, ...}, ...})
        /go/vulnerability_mapper/scanner/grype/grype.go:89 +0x40a
github.com/deepfence/vulnerability_mapper/controller/vulnerability-scan-service.(*VulnerabilityScanService).Scan.func1()
        /go/vulnerability_mapper/controller/vulnerability-scan-service/cveservice.go:97 +0x218
created by github.com/deepfence/vulnerability_mapper/controller/vulnerability-scan-service.(*VulnerabilityScanService).Scan
        /go/vulnerability_mapper/controller/vulnerability-scan-service/cveservice.go:66 +0x52a
```

--- a/vulnerability_mapper/scanner/grype/grype.go
+++ b/vulnerability_mapper/scanner/grype/grype.go
@@ -86,7 +86,7 @@ func PopulateFinalReport(vulnerabilities []byte, c util.Config) (string, error) 
 			cvssScore, overallScore, attackVector = GetCvss(cveCVSSScoreList)
 		}
 
-		msfPoCURL, urls := utils.ExtractExploitPOCURL(match.Vulnerability.URLs)
+		msfPoCURL, urls := utils.ExtractExploitPocUrl(match.Vulnerability.URLs)
 
 		report := scanner.VulnerabilityScanReport{
 			Type:                  "cve",

--- a/vulnerability_mapper/utils/utils.go
+++ b/vulnerability_mapper/utils/utils.go
@@ -54,22 +54,20 @@ func Contains(s []string, str string) bool {
 	return false
 }
 
-func ExtractExploitPOCURL(url []string) (string, []string) {
+func ExtractExploitPocUrl(url []string) (string, []string) {
 	if len(url) == 0 {
 		return "", nil
 	}
+	nonExploitPocUrls := make([]string, len(url))
 	var msfURL string
-	for i, u := range url {
+	for _, u := range url {
 		if strings.Contains(u, metasploitURLPattern) {
 			msfURL = u
-			url = removeFromSlice(url, i)
+		} else {
+			nonExploitPocUrls = append(nonExploitPocUrls, u)
 		}
 	}
-	return msfURL, url
-}
-
-func removeFromSlice(slice []string, s int) []string {
-	return append(slice[:s], slice[s+1:]...)
+	return msfURL, nonExploitPocUrls
 }
 
 func getTimestamp() string {

